### PR TITLE
[AFD] Honksy

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
Honkbots have a cmag interaction now.

(Intended to merge later, too.)

## Why It's Good For The Game
It's funny.

## Images of changes
https://github.com/ParadiseSS13/Paradise/assets/98381/8e069386-3602-4823-8001-0f9944e1a00c
(pre-handcuffs)

## Testing
Cmagged a honkbot. Was restrained with licoriche.

## Changelog
:cl:
add: Honkbots can now be cmagged.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/honkbot.dm [hunk 1]
```diff
@@ -130,6 +130,29 @@
 		playsound(src, 'sound/machines/honkbot_evil_laugh.ogg', 75, 1, -1) // evil laughter
 		update_icon()
 
+/mob/living/simple_animal/bot/honkbot/cmag_act(mob/user)
+	if(HAS_TRAIT(src, TRAIT_CMAGGED))
+		return
+	if(locked || !open)
+		to_chat(user, "<span class='warning'>Unlock and open it with a screwdriver first!")
```
Comment (Burzah) [inline code/modules/mob/living/simple_animal/bot/honkbot.dm:137]:
```suggestion
		to_chat(user, "<span class='warning'>Unlock and open it with a screwdriver first!</span>")
```

File: code/modules/mob/living/simple_animal/bot/honkbot.dm [hunk 2]
```diff
@@ -130,6 +130,29 @@
...
+		return FALSE
+
+	ADD_TRAIT(src, TRAIT_CMAGGED, CLOWN_EMAG)
+	remote_disabled = TRUE
+	locked = TRUE
+	open = FALSE
+	bot_reset()
+	turn_on()
+	if(user)
+		to_chat(user, "<span class='notice'>You smear bananium ooze all over [src]'s circuitry")
```
Comment (Burzah) [inline code/modules/mob/living/simple_animal/bot/honkbot.dm:147]:
```suggestion
		to_chat(user, "<span class='notice'>You smear bananium ooze all over [src]'s circuitry!</span>")
```

**Review by Burzah — CHANGES_REQUESTED** [review]

**Review by Burzah — APPROVED** [review]
Tentative approval. Need an additional reviewer to go over this one.

**Comment by spacegeese** [comment]
this should just be a real feature

**Pull request merged**